src/benchmark_environments/util.py

```diff
@@ -17,3 +22,183 @@ def step(self, action):
+        """
+        if not self.at_absorb_state:
+            inner_obs, inner_rew, done, inner_info = self.env.step(action)
+            if done:
+                # Initialize the artificial absorb state, which we will repeatedly use
+                # starting on the next call to `step()`.
+                self.at_absorb_state = True
+
+                if self.absorb_obs_default is None:
+                    self.absorb_obs_this_episode = inner_obs
+                else:
+                    self.absorb_obs_this_episode = self.absorb_obs_default
+            obs, rew, info = inner_obs, inner_rew, inner_info
+        else:
+            assert self.absorb_obs_this_episode is not None
+            assert self.absorb_reward is not None
+            obs = self.absorb_obs_this_episode
+            rew = self.absorb_reward
+            info = {}  # XXX: Also allow custom info for absorb state.
+
+        return obs, rew, False, info
+
+
+class FixedRewardAfterDoneWrapper(gym.Wrapper):
+    """After the inner environment returns done=True, modify the returned reward.
+
+    Always returns done=False so that the episode can continue until reset().
+    """
+
+    def __init__(self, env: gym.Env, end_reward: float = 0.0):
+        """Initialize wrapper.
+
+        Args:
+            env: The wrapped Env.
+            end_reward: The reward returned after the inner environment returns
+                done=True from `step()`.
+        """
+        super().__init__(env)
+        self.saw_done = None
+        self.end_reward = end_reward
+
+    def reset(self, *args, **kwargs):
+        """Reset the environment."""
+        self.saw_done = False
+        return self.env.reset(*args, **kwargs)
+
+    def step(self, action):
+        """Advance the environment by one step.
+
+        This wrapped `step()` always returns done=False.
+        """
+        obs, rew, done, info = self.env.step(action)
+        if self.saw_done:
+            rew = self.end_reward
+        self.saw_done = self.saw_done or done
+        return obs, rew, False, info
+
+
+def make_env_no_wrappers(env_name: str, **kwargs) -> gym.Env:
+    """Gym sometimes wraps envs in TimeLimit before returning from gym.make().
+
+    This helper method builds directly from step to avoid this wrapper.
+    """
+    return gym.envs.registry.env_specs[env_name].make(**kwargs)
+
+
+def get_gym_max_episode_steps(env_name: str) -> Optional[int]:
+    """Get the `max_episode_steps` attribute associated with a gym Spec."""
+    return gym.envs.registry.env_specs[env_name].max_episode_steps
+
+
+def _gym_register_as_decorator(
+    func: Callable[..., gym.Env],
+    id_base: str,
+    *,
+    module_name: str,
+    id_prefix: str = "",
+    **register_kwargs,
+) -> Callable[..., gym.Env]:
+    """Decorator variant of `gym.register` with autogenerated `id` and `entry_point`.
+
+    (`reward_threshold`, `nondeterministic`, `max_episode_steps`, and `kwargs` are
```

Comment:
I still find this comment a bit confusing (documenting another functions arguments). How about:

```python
`**register_kwargs` is passed through directly to `gym.register`. This function
takes optional arguments `reward_threshold`, `nondeterministic`, `max_episode_steps`,
and `kwargs`, documented below.
```